pkg/cache/context.go

```diff
@@ -595,7 +595,9 @@ func (ctx *Context) getNamespaceObject(namespace string) *v1.Namespace {
 
 func (ctx *Context) AddApplication(request *interfaces.AddApplicationRequest) interfaces.ManagedApp {
 	log.Logger().Debug("AddApplication", zap.Any("Request", request))
-	if app := ctx.GetApplication(request.Metadata.ApplicationID); app != nil {
+	appID := request.Metadata.ApplicationID
+	if app := ctx.GetApplication(request.Metadata.ApplicationID); app != nil && !request.ForceUpdate {
+		log.Logger().Debug("Application already in the context", zap.String("appID", appID))
```

Comment:
This part will likely change after YUNIKORN-1186. With proper coordination, the replacement won't be needed. But in this case, the patch stands on its own.